src/encoder/encoderwave.cpp

```diff
@@ -0,0 +1,214 @@
+        m_sfInfo.format = SF_FORMAT_AIFF;
+    }
+    else {
+        qWarning() << "Unexpected Format when setting EncoderWave: " << format.internalName << ". Reverting to wav";
+        // Other possibly interesting formats 
+        // There is a n option for RF64 to automatically downgrade to RIFF WAV if less than 4GB using an
+        // sf_command, so it could be interesting to use it in place of FORMAT_WAVE.
+        // SF_FORMAT_W64          = 0x0B0000,     /* Sonic Foundry's 64 bit RIFF/WAV */
+        // SF_FORMAT_RF64         = 0x220000,     /* RF64 WAV file */
+
+        // I guess this one is WAVEFORMATEXTENSIBLE, not WAVEFORMATEX.
+        // Not really useful for us since it's mostly for multichannel setups.
+        // SF_FORMAT_WAVEX        = 0x130000,     /* MS WAVE with WAVEFORMATEX */
+
+        // SF_FORMAT_CAF          = 0x180000,     /* Core Audio File format */
+    }
+    int radio = settings.getSelectedOption(EncoderWaveSettings::BITS_GROUP);
+    switch(radio) {
+        case 0:
+            m_sfInfo.format |= SF_FORMAT_PCM_16;
+            break;
+        case 1:
+            m_sfInfo.format |= SF_FORMAT_PCM_24;
+            break;
+        case 2:
+            m_sfInfo.format |= SF_FORMAT_FLOAT;
+            break;
+        default:
+            m_sfInfo.format = SF_FORMAT_WAV | SF_FORMAT_PCM_16;
+            qWarning() << " Unexpected radio index on EncoderWave: " 
+                    << radio << ". reverting to PCM 16bits";
+            break;
+    }
+}
+
+// call sendPackages() or write() after 'flush()' as outlined in enginebroadcast.cpp
+void EncoderWave::flush() {
+    sf_write_sync(m_pSndfile);
+}
+
+
+void EncoderWave::encodeBuffer(const CSAMPLE *pBuffer, const int iBufferSize) {
+    sf_write_float(m_pSndfile, pBuffer, iBufferSize);
+}
+
+/* Originally called from enginebroadcast.cpp to update metadata information
+ * when streaming, however, this causes pops
+ *
+ * Currently this method is used before init() once to save artist, title and album
+*/
+void EncoderWave::updateMetaData(const QString& artist, const QString& title, const QString& album) {
+    m_metaDataTitle = title;
+    m_metaDataArtist = artist;
+    m_metaDataAlbum = album;
+}
+
+void EncoderWave::initStream() {
+
+    // Tell the encoder to automatically convert float input range to the correct output range.
+    sf_command(m_pSndfile, SFC_SET_NORM_FLOAT, NULL, SF_TRUE);
+    // Tell the encoder that, when converting to integer formats, clip
+    // automatically the values that go outside of the allowed range.
+    // Warning! Depending on how libsndfile is compiled autoclip may not work.
+    // Ensure CPU_CLIPS_NEGATIVE and CPU_CLIPS_POSITIVE is setup properly in the build.
+    sf_command(m_pSndfile, SFC_SET_CLIPPING, NULL, SF_TRUE) ;
+
+    // Set meta data. Latin1 since WAVE does not officially support UTF8.
+    int ret = sf_set_string(m_pSndfile, SF_STR_TITLE, m_metaDataTitle.toLatin1().constData());
```

Comment:
This is what says in libsndfile documentation
> Strings passed to and retrieved from these two functions are assumed to be utf-8. However, while formats like Ogg/Vorbis and FLAC fully support utf-8, others like WAV and AIFF officially only support ASCII. Writing utf-8 strings to WAV and AIF files with libsndfile will work when read back with libsndfile, but may not work with other programs.

In the end, we depend on what do the implementations understand. I wanted to tag a wav from windows explorer and it doesn't let you do it (it does for mp3). Tagging it in windows mediaa player didn't save it to the file. (it does for mp3). Maybe the safest bet, as you say, is to simply use toAscii